docs/recipes/MigratingToI18Next.md

```diff
@@ -0,0 +1,164 @@
+// success-line-end
+  return null
+}
+```
+
+This ensures that your app will wait until `react-i18next` is fully initialized before continuing, preventing any issues with missing translations.
+
+## Step 3: Update the i18n initialization method
+
+Next, update your i18n initialization to use `react-i18next`, which also includes RTL (right-to-left) language support and handles locale selection. In a Ingnite generated project, this is located in `app/i18n/i18n.ts`.
+
+```js
+
+import * as i18next from "i18next"
+
+const initI18n = async () => {
+  await i18n.use(initReactI18next).init({
+    resources,
+    lng: pickSupportedLocale()?.languageTag || fallbackLocale,
+    fallbackLng: fallbackLocale,
+    interpolation: { escapeValue: false },
+  });
+
+  const locale = pickSupportedLocale();
+  if (locale?.textDirection === 'rtl') {
+    I18nManager.allowRTL(true);
+    isRTL = true;
+  } else {
+    I18nManager.allowRTL(false);
+    isRTL = false;
+  }
+
+  return i18n;
+};
+```
+
+This ensures that supported locales are chosen based on the device’s settings, and RTL is correctly applied when necessary. For more on detail on these changes, check the [this PR](https://github.com/infinitered/ignite/pull/2770).
+
+## Step 4: Add intl-pluralrules for react-i18next and JSON v4
+
+To support pluralization and `react-i18next`'s JSON v4 format, you’ll need to add the `intl-pluralrules` package:
+
+```bash
+yarn add intl-pluralrules
+```
+
+Make sure to import this package into your i18n configuration file (`app/i18n/i18n.ts`):
+
+```js
+import 'intl-pluralrules';
+```
+
+## Step 5: Update the translate function
+
+The next step is to replace your old translate function with the one provided by `react-i18next`. This is located in `app/i18n/translate.ts`:
+
+```js
+// error-line
+import { TranslateOptions } from "i18n-js"
+// success-line
+import { TOptions } from "i18next"
+
+// error-line-start
+export function translate(key: TxKeyPath, options?: TranslateOptions): string {
+  return i18n.t(key, options)
+// error-line-end
+// success-line-start
+export function translate(key, options) {
+  return i18n.isInitialized ? i18n.t(key, options) : key;
+// success-line-end
+}
+```
+
+## Step 6: Update translation keys from dots (.) to colons (:)
+
+`react-i18next` uses different types of separators for translation keys. Colons (:) are used for first-level translations within an object, while dots (.) are used for nested translations. As a result, you’ll need to update all translation keys in your app accordingly. For example:
+
+```js
+translate("common.ok")
+```
+
+Should be changed to:
+
+```js
+translate("common:ok")
+```
+
+---
+
+Lastly, update the usage of `i18n`s `locale` method to `language` instead. For example in `app/utils/formatDate.ts`:
+
+```js
+// error-line
+const locale = i18n.locale.split("-")[0]
+// success-line
+const locale = i18n.language.split("-")[0]
```

Comment:
```suggestion
const locale = i18next.language.split("-")[0]
```

also needs import of
`import i18next from "i18next"` which would have been in place of `import { i18n } from "app/i18n"`